frontend/src/App.js

```diff
@@ -63,10 +64,14 @@ function App() {
         {/* {process.env.NODE_ENV === 'development' && <Route path="/reduxsandbox" element={<ReduxSandbox />} />} */}
         <Route path="/match-page" element={<MatchPage />} />
         {/* Any routes that require a user to be logged in go here; */}
-        <Route element={<PrivateRoute user={user} />}>
```

Comment:
This route has to wrap all the routes we want to keep "private" / allow only authenticated users to access. The routes nested under this one will only be accessible if a user has loggedIn set true in the UserContext.

It should be like:
<Route element={<PrivateRoute user={user} />}>
    // Private routes go within 
    <Route path="user-profile" element={UserProfile />} />
    <Route path="match-page" element={<MatchPage />} />
</Route> // end protected routes

So if a user not logged in tries to navigate to /user-profile it'll send them back to the guest homepage